rs/nns/cmc/src/main.rs

```diff
@@ -1569,6 +1569,62 @@ fn memo_to_intent_str(memo: Memo) -> String {
     }
 }
 
+/// Returns Ok if transaction matches expected_memo.
+///
+/// First, looks at the memo field. If that does not match, falls back to the
```

Comment:
Nit. Why put implementation detains into doc string? Does the caller of this function need to know how it works?

Reply:
Seems to me, this knowledge is necessary to get desired behavior.

How would you change this?

Reply:
Ah, thanks for clarifying.